### PR TITLE
Fix Graph and ProjectCache UTs

### PR DESCRIPTION
**Pull request by dfederm**
This fixes some broken Graph and ProjectCache UTs (which aren't failing PR or CI builds since they're disabled by #4501...) and also does some minor refactoring/cleanup of the UTs

### CODE REVIEW

**Review by Forgind — APPROVED** [review]
I didn't see anything that should actually change behavior, just cleanup. Did I miss something? (Not counting UTs)

**Comment by dfederm** [comment]
> I didn't see anything that should actually change behavior, just cleanup. Did I miss something? (Not counting UTs)

There is a very minor behavior change in `BuildManager` which fixes the handling of circular dependencies for graph builds. It just fails in the correct way now.

**Comment by Forgind** [comment]
I saw that, but I thought you'd just promoted it out of the exception block below. The build event contexts should be able to be shared.

File: src/Build.UnitTests/ProjectCache/ProjectCacheTests.cs [hunk 1]
```diff
@@ -446,7 +443,7 @@ public static IEnumerable<object[]> SuccessfulGraphsWithBuildParameters
                         yield return new object[]
                         {
                             graph,
-                            ((BuildParameters) buildParameters.First()).Clone()
```
Comment (rainersigwald) [inline src/Build.UnitTests/ProjectCache/ProjectCacheTests.cs:449]: Ok to drop the clone?

**Review by rainersigwald — APPROVED** [review]

File: src/Build.UnitTests/ProjectCache/ProjectCacheTests.cs (hunk 1 above)
Comment (dfederm) [reply src/Build.UnitTests/ProjectCache/ProjectCacheTests.cs:449]: Yup, `MultiProcWithAndWithoutInProcNode` yield returns new objects every time, so no need to clone this.

**Pull request merged**